### DISCUSSION TITLE
Adjusting for covariates with "formula = " in ancombc()

### DISCUSSION THREAD

**Discussion by brynaR** [FrederickHuangLin/ANCOMBC#24] [Q&A] [answered]
Hello,

What is the purpose of the "formula" variable in the function? I had initially assumed it would adjust for covariates similar to ANCOM adj_formula and rand_formula (https://github.com/FrederickHuangLin/ANCOM), however changing the formula doesn't appear to change the results for whatever is input for "group = "

Further, the documentation says "Specifying group is required for detecting structural zeros and performing global test," and, as one would guess, the differentially abundant taxa are different for a factor when it is listed as the group compared to simply being present in the formula. For example, I have factor in my data called "x4." When I had `formula = "x4" + "period"` and `group = "period"` the x4 column in the diff_abn table listed ~10 organisms as "true." When I had `formula = "x4" + "period"` and `group = "x4"` the x4 column in the diff_abn table listed ~30 organisms as "true."

Happy to provide more info if anyone thinks they can help. I just don't understand what the "formula" variable is doing.

Thank you

**Comment by FrederickHuangLin** [comment, marked as answer]
Hi @brynaR,

Thanks for your interest in ANCOMBC!

Per your question, you can treat `formula` as specifying independent variables in a linear model (in log scale, though). Therefore, in your case, if the variable of interest is `x4`, and you specify `formula = "x4 + period"`, it means you are trying to detect differentially abundant taxa with regards to `x4` while adjusting `period` effect. As you can see, it does not matter whether the variable of interest specified in the `formula` is continuous or discrete. However, it is not the case when it comes to the test for structural zeros or the global test. For example, due to the definition of the global test (a taxon is differentially abundant in at least one **group**), a discrete `group` variable is required in the function.

The reason why you got inconsistent results using different `group` variables could due to the implicit data format. For example, `period` might be a continuous variable, but when you specify `group = period`, it will be coerced to a factor. You can try something like:
```
metadata$period = as.factor(metadata$period )
```
before running the `ancombc` function.

Hope it helps!
Huang

**Reply by brynaR** [reply]
Should subject number be part of the formula, then? For example, when I tried ANCOM (https://github.com/FrederickHuangLin/ANCOM), I used `adj_formula = "Sex"` and  `rand_formula = "~ 1 | Subject_number"`. Is there a way to similarly adjust for subject number with ANCOMBC or would I just add it to the formula with the other variables? Or is it not important to add with this function?
Thanks for your help!

**Reply by FrederickHuangLin** [reply]
Actually, adjusting the subject effect means fitting a mixed effects model. As you are aware of this feature in ANCOM, it is still under development in ANCOMBC, and the mixed effects model (or adjusting subject effect) is currently not available in ANCOMBC. We plan to update this feature by this year.

Best,
Huang

**Reply by brynaR** [reply]
In the output, does a negative coefficient indicate that the organism has decreased in abundance? If so, some of the results don't seem to match boxplots I have made using boxplot_abundance()

**Reply by FrederickHuangLin** [reply]
Yes you are right, the negative coefficients stand for negative log fold change so the abundance is decreasing. 

ANCOM-BC methodology aims for detecting changes on absolute abundances instead of observed abundances, while I suppose the box plot is for observed abundances visualization. 

For differences between absolute abundance and observed abundance, you can see another paper of mine: https://www.nature.com/articles/s41522-020-00160-w

Best,
Huang
